sktime/param_est/base.py

```diff
@@ -180,196 +193,290 @@ def fit(self, X):
         #####################################################
-        self._fit(X=X_inner)
+        self._fit(X=X_inner, y=y_inner)
 
         # this should happen last
         self._is_fitted = True
 
         return self
 
-    def update(self, X):
+    def update(self, X, y=None):
         """Update fitted parameters on more data.
 
-        If no estimator-specific update method has been implemented,
-        default fall-back is fitting to all observed data so far
-
-        State required:
-            Requires state to be "fitted".
-
-        Accesses in self:
-            Fitted model attributes ending in "_".
-            Pointers to seen data, self._X
-            self._is_fitted
-            model attributes ending in "_".
-
-        Writes to self:
-            Update self._X with `X`, by appending rows.
-            Updates fitted model attributes ending in "_".
+        This method allows for updating the estimator with additional observations.
+        If a custom update method is not implemented, the default behavior is to refit
+        the estimator using all data seen so far. Note that the primary parameter is y;
+        if X is provided, it is used to update the stored feature data.
 
         Parameters
         ----------
-        X : time series in sktime compatible data container format
-                Time series to which to fit the forecaster in the update.
-            y can be in one of the following formats, must be same scitype as in fit:
-            Series scitype: pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
-            Panel scitype: pd.DataFrame with 2-level row MultiIndex,
-                3D np.ndarray, list of Series pd.DataFrame, or nested pd.DataFrame
-            Hierarchical scitype: pd.DataFrame with 3 or more level row MultiIndex
-            For further details:
-                on usage, see forecasting tutorial examples/01_forecasting.ipynb
-                on specification of formats, examples/AA_datatypes_and_datasets.ipynb
+        X : sktime-compatible container, optional (default=None)
+            New time series data (features) for updating the estimator. If not provided,
+            only y is updated.
+        y : array-like
+            New target values for updating the estimator. An empty or None y will result
+            in no update and a warning will be issued.
 
         Returns
         -------
         self : reference to self
         """
         self.check_is_fitted()
 
-        if X is None or (hasattr(X, "__len__") and len(X) == 0):
+        if y is None or (hasattr(y, "__len__") and len(y) == 0):
             warn(
                 f"empty y passed to update of {self}, no update was carried out",
                 obj=self,
             )
             return self
 
-        # input checks and minor coercions on X, y
-        X_inner = self._check_X(X=X)
-
-        # update internal X with the new X
-        self._update_X(X_inner)
+        if X is not None:
```

Comment:
I do not think we need to update `y`, it is not mandatory.

Reply:
so remove it completely?